qiita_db/support_files/patches/python_patches/14.py

```diff
@@ -0,0 +1,20 @@
+# Feberuary 7, 2015
+# This patch recreates all the QIIME mapping files to avoid lower-upper case
+# problems. See https://github.com/biocore/qiita/issues/799
+
+from qiita_db.util import get_mountpoint
+from qiita_db.sql_connection import SQLConnectionHandler
+from qiita_db.metadata_template import PrepTemplate
+
+conn_handler = SQLConnectionHandler()
+
+_id, fp_base = get_mountpoint('templates')[0]
+
+for prep_template_id in conn_handler.execute_fetchall(
+        "SELECT prep_template_id FROM qiita.prep_template"):
+    prep_template_id = prep_template_id[0]
+    pt = PrepTemplate(prep_template_id)
+    study_id = pt.study_id
+
+    for _, fpt in pt.get_filepaths():
+        pt.create_qiime_mapping_file(fpt)
```

Comment:
OK, thanks. I still think we are not modifying any of the sample names,
if you look at the code in `create_qiime_mapping_file`, you'll see it
only was changing the letter-case for the sample ids in this file,
everything else in the rest of the system should be done with the
correct sample identifiers. If not, this PR is not even aware of those
problems. Please let me know if I'm missing something terribly obvious
here.

On (Feb-10-15|13:54), josenavas wrote:
> > +from qiita_db.util import get_mountpoint
> > +from qiita_db.sql_connection import SQLConnectionHandler
> > +from qiita_db.metadata_template import PrepTemplate
> > +
> > +conn_handler = SQLConnectionHandler()
> > +
> > +_id, fp_base = get_mountpoint('templates')[0]
> > +
> > +for prep_template_id in conn_handler.execute_fetchall(
> > +        "SELECT prep_template_id FROM qiita.prep_template"):
> > +    prep_template_id = prep_template_id[0]
> > +    pt = PrepTemplate(prep_template_id)
> > +    study_id = pt.study_id
> > +
> > +    for _, fpt in pt.get_filepaths():
> > +        pt.create_qiime_mapping_file(fpt)
> 
> yes if we are changing the samples that are already on the database...
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/829/files#r24454532